File: models/user.model.js
const db = require('../database/connection');

// Get all users
const getAllUsers = (callback) => {
  const query = 'SELECT * FROM users';
  db.query(query, (err, results) => {
    if (err) return callback(err, null);
    callback(null, results);
  });
};

// Get user by ID
const getUserById = (id, callback) => {
  const query = 'SELECT * FROM users WHERE id = ?';
  db.query(query, [id], (err, results) => {
    if (err) return callback(err, null);
    callback(null, results[0]);
  });
};

// Create user
const createUser = (user, callback) => {
  const query = 'INSERT INTO users (nombre, email, telefono, fecha_creacion) VALUES (?, ?, ?, NOW())';
  db.query(query, [user.nombre, user.email, user.telefono], (err, results) => {
    if (err) return callback(err, null);
    callback(null, results);
  });
};

// Update user
const updateUser = (id, user, callback) => {
  const query = 'UPDATE users SET nombre = ?, email = ?, telefono = ? WHERE id = ?';
  db.query(query, [user.nombre, user.email, user.telefono, id], (err, results) => {
    if (err) return callback(err, null);
    callback(null, results);
  });
};

// Delete user
const deleteUser = (id, callback) => {
  const query = 'DELETE FROM users WHERE id = ?';
  db.query(query, [id], (err, results) => {
    if (err) return callback(err, null);
    callback(null, results);
  });
};

module.exports = {
  getAllUsers,
  getUserById,
  createUser,
  updateUser,
  deleteUser
};

File: models/user_model.py
from psycopg2 import errors

from database.db import get_connection


class DuplicateEmailError(Exception):
    pass


def get_all_users():
    with get_connection() as conn:
        with conn.cursor() as cur:
            cur.execute(
                "SELECT id, full_name, email, created_at FROM users ORDER BY id ASC"
            )
            return cur.fetchall()


def create_user(full_name: str, email: str):
    try:
        with get_connection() as conn:
            with conn.cursor() as cur:
                cur.execute(
                    "INSERT INTO users (full_name, email) VALUES (%s, %s) RETURNING id, full_name, email, created_at",
                    (full_name, email),
                )
                return cur.fetchone()
    except errors.UniqueViolation as exc:
        raise DuplicateEmailError("email already exists") from exc


def update_user(user_id: int, full_name: str, email: str):
    try:
        with get_connection() as conn:
            with conn.cursor() as cur:
                cur.execute(
                    "UPDATE users SET full_name = %s, email = %s WHERE id = %s RETURNING id, full_name, email, created_at",
                    (full_name, email, user_id),
                )
                return cur.fetchone()
    except errors.UniqueViolation as exc:
        raise DuplicateEmailError("email already exists") from exc


def delete_user(user_id: int) -> bool:
    with get_connection() as conn:
        with conn.cursor() as cur:
            cur.execute("DELETE FROM users WHERE id = %s RETURNING id", (user_id,))
            return cur.fetchone() is not None
